Some more samples and fixes in samples

TillAlex opened the pull request:
As Vulkan code should be mostly platform independent platform dependent and independent code in samples was separated (0152794). This allows implementing the same samples for multiple platforms without code doubling.

Until now the XLogoSample and the ClearSample are implemented for Android and Windows (bb4fec3). As soon as MoltenVK is supported I will add iOS samples as well.

While testing the samples under Windows and Android I noticed that they are buggy. Testing with LunarG SDK shows multiple validation errors. 5801bb5 fixes those.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/mono/VulkanSharp?pullRequest=69) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/mono/VulkanSharp?pullRequest=69) before we can accept your contribution.<br/><br/>:x: TillAlex [sign now](https://cla.opensource.microsoft.com/mono/VulkanSharp?pullRequest=69)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/mono/VulkanSharp?pullRequest=69) it.</sub>

TillAlex commented:
Not sure why the CLA bot asks my to sign again. Already signed the agreement in #17

TillAlex commented:
Signed it again now...

radekdoulik commented:
We have new CLA bot as the old one was deprecated, sorry for the inconvenience.

radekdoulik commented on samples/ClearView.Android/MainActivity.cs:
```diff
@@ -15,7 +17,7 @@ protected override void OnCreate (Bundle savedInstanceState)
 
 			LinearLayout layout = FindViewById<LinearLayout> (Resource.Id.Layout);
-			layout.AddView (new ClearView (ApplicationContext));
+			layout.AddView (new VulkanViewSample(ApplicationContext, new ClearSample()));
```
Could you please keep the original indentation here and in other places? (the missing space)

TillAlex replied:
I do not see this style very often and it's hard to enforce it. Visual Studio removes those spaces every time you copy and paste or even type the semicolon at the end of a line.

Should I use spaces before every brace?

radekdoulik replied:
We use coding style explained here http://www.mono-project.com/community/contributing/coding-guidelines/

I wonder whether this might help you in VS, https://blogs.msdn.microsoft.com/dotnet/2016/12/15/code-style-configuration-in-the-vs2017-rc-update/

radekdoulik commented on samples/Samples.Android/VulkanViewSample.cs:
```diff
@@ -0,0 +1,34 @@
+namespace Samples.Android
+{
+    public class VulkanViewSample : VulkanView
+    {
```
Please use tabs for indentation as the rest of the code.

radekdoulik commented on samples/Samples.Common/XLogoSample.cs:
```diff
@@ -279,7 +152,7 @@ Pipeline[] CreatePipelines ()
 			var multisampleCreateInfo = new PipelineMultisampleStateCreateInfo {
 				RasterizationSamples = SampleCountFlags.Count1,
-				SampleMask = new uint [] { ~0u }
+				//SampleMask = new uint [] { ~0u }
```
If setting of this property is not needed, we should just remove that line.

radekdoulik commented on samples/Samples.Common/XLogoSample.cs:
```diff
@@ -318,7 +191,7 @@ Pipeline[] CreatePipelines ()
 				InputAssemblyState = inputAssemblyStateCreateInfo,
 				VertexInputState = vertexInputStateCreateInfo,
-				DynamicState = new PipelineDynamicStateCreateInfo (),
+				//DynamicState = new PipelineDynamicStateCreateInfo (),
```
Same as above.

TillAlex commented:
Just found that I checked in SampleBase in the last commit although there are only white space fixes in that file. If you want me to I can prepare another pull request without that changes.

radekdoulik commented:
No need for another PR. Just add more commits here and I will squash it before merge, or force push updated commits to your branch.

TillAlex commented:
Just added another commit. I fixed lots of white space issues, hope I found them all. I changed the directory structure according to the AnalogClock example.
